File: source-connector/src/main/java/sourceconnector/domain/processor/impl/ByPassProcessor.java
package sourceconnector.domain.processor.impl;

import sourceconnector.domain.log.Log;
import sourceconnector.domain.processor.BaseProcessor;

/**
 * Bypass the input with no operation
 */
public class ByPassProcessor extends BaseProcessor<Log> {
  @Override
  public Log process(Log input) {
    if(this.next == null) return input;
    return this.next.process(input);
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/processor/impl/ByPassProcessor.kt
package sourceconnector.domain.processor.impl

import sourceconnector.domain.log.Log
import sourceconnector.domain.processor.BaseProcessor

/**
 * Bypass the input with no operation
 */
class ByPassProcessor : BaseProcessor<Log>() {
  override fun process(input: Log): Log? {
    if (this.next == null) return input
    return this.next!!.process(input)
  }
}
